Tools/autotest/sim_vehicle.py

```diff
@@ -397,16 +470,17 @@ def default_waf_target(vehicle):
 def options_for_frame(frame, vehicle, opts):
     """Return informatiom about how to sitl for frame e.g. build-type==sitl"""
     ret = None
-    if frame in _options_for_frame:
-        ret = _options_for_frame[frame]
+    frames = _options[vehicle]["frames"]
+    if frame in frames:
+        ret = _options[vehicle]["frames"][frame]
     else:
         for p in ["octa", "tri", "y6", "firefly", "heli", "gazebo", "last_letter", "jsbsim", "quadplane", "plane-elevon", "plane-vtail", "plane"]:
             if frame.startswith(p):
-                ret = _options_for_frame[p]
+                ret = _options[vehicle]["frames"][p]
                 break
     if ret is None:
         if frame.endswith("-heli"):
-            ret = _options_for_frame["heli"]
+            ret = _options[vehicle]["frames"]["heli"]
```

Comment:
Maybe.  But given we're going to have a severe cross-product issue with frame shapes into the future, we might also need to find a better way forward than just a simple "frame"...